wlroots/wlr_types/cursor.py

```diff
@@ -146,6 +141,52 @@ def attach_input_device(self, input_device: InputDevice) -> None:
 
         lib.wlr_cursor_attach_input_device(self._ptr, input_device._ptr)
 
+    def detach_input_device(self, input_device: InputDevice) -> None:
+        """Detach this input device from this cursor
+
+        The input device type must be one of:
+            - InputDeviceType.POINTER
+            - InputDeviceType.TOUCH
+            - InputDeviceType.TABLET_TOOL
+
+        :param input_device:
+            The input device to detach from the cursor
+        """
+        allowed_device_types = (
+            InputDeviceType.POINTER,
+            InputDeviceType.TOUCH,
+            InputDeviceType.TABLET_TOOL,
+        )
+        if input_device.device_type not in allowed_device_types:
+            raise ValueError(
+                f"Input device must be one of pointer, touch, or tablet tool, got: {input_device.device_type}"
+            )
+
+        lib.wlr_cursor_detach_input_device(self._ptr, input_device._ptr)
+
+    def set_image(
+        self,
+        pixels: int | None,
+        stride: int,
```

Comment:
If we allocate a contiguous image block, I don't think we'd need to have `stride` passed in, it would just always be 1?

Reply:
> If we allocate a contiguous image block, I don't think we'd need to have `stride` passed in, it would just always be 1?

To be honest I'm not entirely sure but since the function requires a stride anyways, won't hurt to leave it in ?